xapian-applications/omega/query.cc

```diff
@@ -651,25 +652,38 @@ html_strip(const string &str)
     return res;
 }
 
-// FIXME split list into hash or map and use that rather than linear lookup?
-static int word_in_list(const string& word, const string& list)
-{
-    string::size_type split = 0, split2;
-    int count = 0;
-    while ((split2 = list.find('\t', split)) != string::npos) {
-	if (word.size() == split2 - split) {
-	    if (memcmp(word.data(), list.data() + split, word.size()) == 0)
-		return count;
+static unordered_map<string, int> word_to_occurrence;
+
+class WordList {
+    string list;
+  public:
+    WordList() : list("") { }
```

Comment:
There's no reason to use `list("")`, and it's actually a bit less efficient with current compilers (or at least for 4 year old compilers - I haven't rechecked this recently, but see: https://survex.com/~olly/blog/coding/empty-strings-in-c%2B%2B.html)

In C++, variables with a class type are always default constructed (unlike variables of fundamental types, such as int and double) and a default constructed `std::string` is an empty string, so there's no need to initialise `list` explicitly at all here.  If you **really** want to then `list()` should be exactly equivalent to the implicit initialisation which happens anyway.